### PR TITLE
Patient Entity Management contract

### PR DESCRIPTION
**Pull request by Emeka000**
Pull Request Overview
Summary

This PR implements a smart contract to manage patient identities on-chain. It covers patient profile creation, demographic information storage, and optional links to insurance and medical history references.
Related Issues

Closes #[29]

Type of Change

Mark with an x all the checkboxes that apply (like [x]).

[ x] Feat (A new feature)
Fix (A bug fix)
Docs (Documentation changes)
Style (changes that do not affect the meaning of the code: formatting, etc)
[ x] Refactor (code changes that neither fix a bug nor add a feature)
[ x] Test (adding missing tests or correcting existing tests)
Build (changes that affect the build system or external dependencies)
Chore (maintenance changes that do not fall into any of the other categories)

Changes Made

Developed core patient management functions: register_patient, update_patient, and get_patient.

Designed the PatientData structure for efficient on-chain storage.

Added basic tests to validate patient registration and updates.
Technical Notes

Patient metadata is stored as a string to allow flexible external referencing (e.g., off-chain insurance records, hospital records).

All wallet address mappings are handled securely to avoid collisions.

✅ Tests Results

Successfully registered a new patient.

Successfully updated patient metadata.

Successfully retrieved patient data based on wallet address.
Test Coverage

✅ Unit Tests
📨 Integration Tests
📟 Manual Testing

Evidence

Running tests...
✔️ register_patient: PASSED
✔️ update_patient: PASSED
✔️ get_patient: PASSED
Testing Notes

Tested edge cases: double registration, updating non-existent patient records.

Confirmed that metadata updates overwrite the previous entry safely.
Next Steps

Extend metadata validation.

Integrate off-chain verification methods for linked insurance/medical history metadata.

Add access control for sensitive patient updates.

### CODE REVIEW

File: apps/vitacare-contracts/contracts/ Patient Entity Management Contract/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+mod patient_identity {
+    use ink_storage::{
+        collections::HashMap,
+        traits::{PackedLayout, SpreadLayout},
+    };
+
+    /// Defines the patient data structure
+    #[derive(Debug, Clone, scale::Encode, scale::Decode, SpreadLayout, PackedLayout)]
+    #[cfg_attr(feature = "std", derive(scale_info::TypeInfo))]
+    pub struct PatientData {
```
Comment (titito2005) [inline apps/vitacare-contracts/contracts/ Patient Entity Management Contract/src/lib.rs:15]: Please create a types file

File: apps/vitacare-contracts/contracts/ Patient Entity Management Contract/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+                metadata,
+                registered_at: now,
+                last_updated_at: now,
+            };
+
+            // Store the patient data
+            self.patients.insert(wallet, patient.clone());
+
+            // Emit event
+            self.env().emit_event(PatientRegistered {
```
Comment (titito2005) [inline apps/vitacare-contracts/contracts/ Patient Entity Management Contract/src/lib.rs:108]: Please create a storage file

File: apps/vitacare-contracts/contracts/ Patient Entity Management Contract/src/lib.rs (hunk 1 above)
Comment (titito2005) [inline apps/vitacare-contracts/contracts/ Patient Entity Management Contract/src/lib.rs:15]: Create a types file

**Review by titito2005 — CHANGES_REQUESTED** [review]
Please follow this example:
https://github.com/VitaCare-Organization/vitacare-dev/pull/42
